network/oauth_client.go

```diff
@@ -39,9 +40,7 @@ func NewOAuthClient(target, username, password string, insecureSkipVerify bool)
 				Timeout:   5 * time.Second,
 				KeepAlive: 30 * time.Second,
 			}).Dial,
-			ResponseHeaderTimeout: 15 * time.Minute,
```

Comment:
We added the ResponseHeaderTimeout because we wanted to make sure that we also enforced that. This is separate from the overall timeout, which, as replied below, we believe is respected. We are OK with removing it since we agree that it should cover the entire lifecycle of the connection, but if we see issues related to this we may add it back.
https://blog.cloudflare.com/the-complete-guide-to-golang-net-http-timeouts/